### PR TITLE
MOTECH-1893 Create additional IT tests for cascade delete

### PR DESCRIPTION
**Pull request by atish160384**
IT tests for behavior of cascade delete for Uni Directional, Bidirectional, Map Relationship.

### CODE REVIEW

**Comment by pgesek** [comment]
@sebbrudzinski can you review this?

File: platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java [hunk 1]
```diff
@@ -500,6 +543,90 @@ public void testCascadeDelete() {
...
+        assertNotNull(mother);
+
+        ChildCase child = childCaseDataService.findById(child1.getId());
+        assertNotNull(child);
+
+        motherCaseDataService.delete(mother);
+        mother = motherCaseDataService.findById(mother.getId());
+        assertNull(mother);
+
+        child = childCaseDataService.findById(child2.getId());
```
Comment (sebbrudzinski) [inline platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java]: Why don't we also verify that another child was deleted?

File: platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java [hunk 2]
```diff
@@ -500,6 +543,90 @@ public void testCascadeDelete() {
     }
 
     @Test
+    public void testCascadeDeleteUniDirectional() {
+        ChildCase child1 = new ChildCase("Child 1");
+        childCaseDataService.create(child1);
+
+        ChildCase child2 = new ChildCase("Child 2");
+        childCaseDataService.create(child2);
```
Comment (sebbrudzinski) [inline platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java]: Side note - it shouldn't be required to manually create related objects. They should be by default created with the owning instance (mother).

File: platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java [hunk 3]
```diff
@@ -500,6 +543,90 @@ public void testCascadeDelete() {
...
+
+        event = eventDataService.findById(event.getId());
+        assertNotNull(event);
+
+        eventLogDataService.delete(eventLog);
+        eventLog = eventLogDataService.findById(eventLog.getId());
+        assertNull(eventLog);
+
+        event = eventDataService.findById(event.getId());
+        assertNull(event);
```
Comment (sebbrudzinski) [inline platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java]: This doesn't seem to check the map at all, even though the test name says so. It checks cascade delete between EventLog and Event, which is unidirectional relationship.

File: platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java [hunk 4]
```diff
@@ -500,6 +543,90 @@ public void testCascadeDelete() {
...
+
+        Unit unit = new Unit("Tree");
```
Comment (sebbrudzinski) [inline platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java]: Seems like all of the below belongs to a new test. Also not sure why are we testing unidirectional relationships again.

**Comment by sebbrudzinski** [comment]
Besides the inline comments, I'm still not sure if we already test all cases. Can we refactor test names for cascade deletes to describe the test cases? We should test one-to-one, one-to-many (uni- and bi- directional) and many-to-many relationships and of course maps.

**Pull request closed without merging**